src/recipes.jl

```diff
@@ -378,109 +378,256 @@ end
 end
 @deps bar shape
 
+
 # ---------------------------------------------------------------------------
 # Histograms
 
-# edges from number of bins
-function calc_edges(v, bins::Integer)
-    vmin, vmax = extrema(v)
-    linspace(vmin, vmax, bins+1)
+_bin_centers(v::AVec) = (v[1:end-1] + v[2:end]) / 2
```

Comment:
> Also, is there any reason for removing the centers function (below) and defining a new bin_centers function? This seems unnecessary

`centers` is only used here, and it's not exported - so I wanted to clearly mark it as an internal function and make the name a bit more descriptive. I don't think would be obvious to someone new who just reads `centers(v)` what the function does.